### DISCUSSION TITLE
make install on Apple

### DISCUSSION THREAD

**Discussion by nenanth** [snl-dakota/dakota#26] [Q&A] [answered]
Hi everyone - my first post on this board! 
I'm having trouble with the last step in the setup process for Dakota: `make install`

I built dakota on my apple laptop (intel, 2020) with the following settings
1. gcc/g++ version 9.5.0
1. Boost v1.76.0, built from source with gcc-9 (using c++11)
2. Lapack/blas v3.11.0, built from source with gcc-9
3. Dakota 6.17.0, built from source with gcc-9 (and pointing to the boost and lapack from steps 1 and 2)
4. CMake v3.17.5 (downgraded from 3.25, no real difference to error messages for this one change)
 
At the last step, for `make install`, most of the installation occurs successfully.. but I get some CMake error that looks like this:
```
CMake Error at /Users/ananthsridharan/Downloads/build/dakota-6.17.0-release-public-src-cli/cmake/InstallDarwinDylibs.cmake:89 (file):
  file Resolved path is not absolute
Call Stack (most recent call first):
  scripts/cmake_install.cmake:53 (include)
  cmake_install.cmake:86 (include)
```

The appropriate section of the cmake file is 
```
file(GET_RUNTIME_DEPENDENCIES
  RESOLVED_DEPENDENCIES_VAR dakota_exe_deps
  UNRESOLVED_DEPENDENCIES_VAR dakota_exe_deps_unresolved
  EXECUTABLES "${CMAKE_CURRENT_BINARY_DIR}/src/dakota"
  LIBRARIES "${CMAKE_CURRENT_BINARY_DIR}/src/libdakota_src_fortran.dylib" "${CMAKE_CURRENT_BINARY_DIR}/src/libdakota_src.dylib"
  POST_EXCLUDE_REGEXES "/system" "/usr/lib" "/usr/XLL" "${CMAKE_CURRENT_BINARY_DIR}"
)
```

Not sure how to debug the intermediate outputs of this command, because it fails before completing.. any ideas?

**Comment by jadamstephens** [comment, marked as answer]
This is a relatively new CMake command for us (used for the first time in Dakota 6.17), and I've never seen this error ("file Resolved path is not absolute").

The code is intended to identify all the dependencies of the dakota executable and two of the top level Dakota libraries so that their RPATHs can be updated correctly at install time (that is, so that Dakota can find all of the libraries it needs when it's run).

The easiest possible fix is that you comment out that file command from InstallDarwinDylibs.cmake, and also the following list append command (lines 84-92). These commands are not needed on every machine, and yours might be one where they aren't. You'll find out whether it worked when you try to run Dakota from the install location. If it complains about missing libraries, we'll be back to the drawing board.

**Reply by nenanth** [reply]
thanks for the quick reply. dakota runs just fine even after the install terminates abruptly (and the bin directory is not as fully populated as the prebuilt versions). I'll comment it out and see if there are any more hiccups.. for now, I'm good!
